File: cpj-backend/src/main/java/br/com/ota/cpjbackend/model/Expense.java
package br.com.ota.cpjbackend.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.math.BigDecimal;
import java.util.Date;

@Data
@Entity
@Table(name = "expenses")
@NoArgsConstructor
public class Expense extends BaseEntity {

    @NotBlank
    @Size(max = 200)
    private String description;

    private Date date;

    @NotNull
    private BigDecimal paymentValue;

    private boolean refundable;

    @ManyToOne(optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Contract contract;

}

File: cpj-frontend/src/models/expense.js
import Contract from "./contract";

export default class Expense {

    constructor(date) {
        this.description = '';
        this.date = date;
        this.paymentValue = '';
        this.contract = new Contract();
        this.refundable = true;
    }

    defaults() {
        return {
            description: '',
            date: new Date(),
            paymentValue: '',
            contract: new Contract(),
            refundable: false,
            payed: false,
        }
    }


}